### ISSUE TITLE
e_ext_ian_uri_host_not_fqdn_or_ip: false positive on valid URIs (same defect fixed for the SAN lint in #199)

### ISSUE THREAD

**Issue by glaziermag** [zmap/zlint#1081]
`e_ext_ian_uri_host_not_fqdn_or_ip` returns `error` for issuerAltName URIs that conform to RFC 5280 in two cases: (a) URIs with no authority component (e.g. `urn:isbn:0451450523`, `sip:alice@sip.uri.com`), and (b) URIs whose authority host is a bracketed IPv6 literal (e.g. `https://[2001:db8::1]/`). This is the same defect class reported for the SAN twin lint in #198 and fixed by #199 (2018) — but #199 touched only the SAN lint, its fixtures, and `util/fqdn.go`, so the IAN lint still runs the pre-#199 logic. `e_ext_ian_uri_host_not_fqdn_or_ip` is now the only caller of `util.AuthIsFQDNOrIP` in the tree.

## Expected behavior

RFC 5280, § 4.2.1.6 (the text the lint's own comment block and `Description` quote):

> The name MUST include both a scheme (e.g., "http" or "ftp") and a scheme-specific-part. URIs that include an authority ([RFC3986], Section 3.2) MUST include a fully qualified domain name or IP address as the host.

The MUST is conditional on the URI *including an authority*, and an IP-address host is expressly permitted. § 4.2.1.7 applies the same rule to issuerAltName: "Issuer alternative name MUST be encoded as in 4.2.1.6." So:

- `urn:isbn:0451450523` and `sip:alice@sip.uri.com` have a scheme and scheme-specific-part but no authority → vacuously conformant. (`sip:alice@sip.uri.com` is the exact URI in your own SAN fixture `v3/testdata/SANURINoAuthority.pem`, which `TestSANURINoAuthority` expects to `Pass`.)
- `https://[2001:db8::1]/` has an authority whose host *is* an IP address → conformant.

## Observed behavior

At master (7d02b26), `zlint -includeNames=e_ext_ian_uri_host_not_fqdn_or_ip <cert>` on self-signed certs (NotBefore 2024-01-01, i.e. after the lint's `RFC5280Date` effective date) carrying a single issuerAltName uniformResourceIdentifier:

| issuerAltName URI | `e_ext_ian_uri_host_not_fqdn_or_ip` | SAN twin on same URI in SAN |
|---|---|---|
| `urn:isbn:0451450523` | **error** | pass |
| `sip:alice@sip.uri.com` | **error** | pass |
| `https://[2001:db8::1]/` | **error** | pass |
| `https://example.com/` (control) | pass | pass |

## Root cause

Two mechanisms:

1. [`lint_ext_ian_uri_host_not_fqdn_or_ip.go#L61-L72`](https://github.com/zmap/zlint/blob/7d02b26ecaf0d315d64d00d010b1cd8a78d0039c/v3/lints/rfc/lint_ext_ian_uri_host_not_fqdn_or_ip.go#L61-L72): `Execute` feeds `parsedUrl.Host` to `util.AuthIsFQDNOrIP` unconditionally. For opaque URIs `url.Parse` yields `Opaque="isbn:0451450523"` / `"alice@sip.uri.com"` with `Host=""`, and the empty host fails the check. The SAN twin has the guard #199 added ([`lint_ext_san_uri_host_not_fqdn_or_ip.go#L68`](https://github.com/zmap/zlint/blob/7d02b26ecaf0d315d64d00d010b1cd8a78d0039c/v3/lints/rfc/lint_ext_san_uri_host_not_fqdn_or_ip.go#L68): `if parsed.Opaque == ""`, "no authority … vacuously OK").
2. [`util/fqdn.go#L81`](https://github.com/zmap/zlint/blob/7d02b26ecaf0d315d64d00d010b1cd8a78d0039c/v3/util/fqdn.go#L81): `util.GetHost` does `end := strings.Index(auth, ":")`, truncating `[2001:db8::1]` to `[2001`, which then fails `IsFQDNOrIP`. (The SAN twin does not call `GetHost`; it passes `parsed.Host` to `IsFQDNOrIP` directly.)

The IAN lint's test fixtures (`IANURIHost{NotFQDNOrIP,FQDN,IP,WildcardFQDN,WrongWildcard,Asterisk}.pem`) cover only FQDN/IPv4/wildcard hosts, so neither case is exercised — the no-authority and IPv6 paths have no IAN-side coverage, while the SAN side gained `SANURINoAuthority.pem` in #199.

## Suggested fix

Mirror the SAN lint's post-#199 `Execute` body over `c.IANURIs`, and add IAN fixtures for the missing cases (opaque URI, `sip:user@host`, bracketed IPv6). I'm happy to send a small PR doing exactly that if useful.

## Repro certs

<details><summary>ian_urn_isbn.pem</summary>

```
-----BEGIN CERTIFICATE-----
MIIBazCCARCgAwIBAgICMDkwCgYIKoZIzj0EAwIwLDEOMAwGA1UEChMFUmVwcm8x
GjAYBgNVBAMTEXJlcHJvLmV4YW1wbGUuY29tMB4XDTI0MDEwMTAwMDAwMFoXDTI0
MTIzMTAwMDAwMFowLDEOMAwGA1UEChMFUmVwcm8xGjAYBgNVBAMTEXJlcHJvLmV4
YW1wbGUuY29tMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAELmxYb8PXGNkJju85
FUuJUEp9KwNWzUW12TMvtwMMABPmqIh5rzKm7BPkV0cedOMQBh3xgFO4DMCnFkBJ
QdJOlqMiMCAwHgYDVR0SBBcwFYYTdXJuOmlzYm46MDQ1MTQ1MDUyMzAKBggqhkjO
PQQDAgNJADBGAiEA5QqURSdYZFLfrr/95ijzGQHatSrxP37jarGB+i87WQcCIQD/
WLsdZTNyjfeip010QxzJyZuMwoM8MSA1AaA7n4mKEQ==
-----END CERTIFICATE-----
```

</details>

<details><summary>ian_sip_alice.pem</summary>

```
-----BEGIN CERTIFICATE-----
MIIBazCCARKgAwIBAgICMDkwCgYIKoZIzj0EAwIwLDEOMAwGA1UEChMFUmVwcm8x
GjAYBgNVBAMTEXJlcHJvLmV4YW1wbGUuY29tMB4XDTI0MDEwMTAwMDAwMFoXDTI0
MTIzMTAwMDAwMFowLDEOMAwGA1UEChMFUmVwcm8xGjAYBgNVBAMTEXJlcHJvLmV4
YW1wbGUuY29tMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE53s6xFOdtTDqCS2W
U3I+BS8kiOhsahnneVgY2YP0ymTPhotkPjFS2ocuZjNn+8bUaYXbSi2Fz6En5ILD
zm9Hl6MkMCIwIAYDVR0SBBkwF4YVc2lwOmFsaWNlQHNpcC51cmkuY29tMAoGCCqG
SM49BAMCA0cAMEQCIBhFYNOJuLz4UlY1JcmsTGVZZ6+JU6XDsJYo7gkRIbFBAiBo
Fhl9TXBtKJ4cOmBwprxAXqjgnX7qwMlyI0iPMDOxUg==
-----END CERTIFICATE-----
```

</details>

<details><summary>ian_ipv6.pem</summary>

```
-----BEGIN CERTIFICATE-----
MIIBbTCCAROgAwIBAgICMDkwCgYIKoZIzj0EAwIwLDEOMAwGA1UEChMFUmVwcm8x
GjAYBgNVBAMTEXJlcHJvLmV4YW1wbGUuY29tMB4XDTI0MDEwMTAwMDAwMFoXDTI0
MTIzMTAwMDAwMFowLDEOMAwGA1UEChMFUmVwcm8xGjAYBgNVBAMTEXJlcHJvLmV4
YW1wbGUuY29tMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEh3/p0HcOR6p3tXHk
5c0OiKOxT4jXwQ0szo7456hEPjJ5sIpQIzyTqNeKk1xr2f6T/rdo9QTooqt2sakZ
RAI0XaMlMCMwIQYDVR0SBBowGIYWaHR0cHM6Ly9bMjAwMTpkYjg6OjFdLzAKBggq
hkjOPQQDAgNIADBFAiBqcc2j0kyJS/ngdUQT9kXHuAyXRj4RZpfOEjjOSPC4+AIh
AJq4ecC2hGTXwBJZf6BWkaIA/KkcbB+paGhXHwq5TD/i
-----END CERTIFICATE-----
```

</details>

<details><summary>ian_control_https_example_com.pem</summary>

```
-----BEGIN CERTIFICATE-----
MIIBazCCARGgAwIBAgICMDkwCgYIKoZIzj0EAwIwLDEOMAwGA1UEChMFUmVwcm8x
GjAYBgNVBAMTEXJlcHJvLmV4YW1wbGUuY29tMB4XDTI0MDEwMTAwMDAwMFoXDTI0
MTIzMTAwMDAwMFowLDEOMAwGA1UEChMFUmVwcm8xGjAYBgNVBAMTEXJlcHJvLmV4
YW1wbGUuY29tMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAECbuZMHjaiii/ewfJ
1ggXkRYOaGo2pVAvLssfF0PRoSYmrlGXPpbKkdQ6FkQbxXLVmfHWb4W0aM/m/cFX
b07fBaMjMCEwHwYDVR0SBBgwFoYUaHR0cHM6Ly9leGFtcGxlLmNvbS8wCgYIKoZI
zj0EAwIDSAAwRQIhAMVDA6f7rvNrqbtKxhNugAS2HoHOzlVtFiMZTtLZWPMSAiBd
iqXe/Vk6DxIl/Z3xF448vB3vazDFXcGrvihV46cTGw==
-----END CERTIFICATE-----
```

</details>

Certs were generated with Go's `crypto/x509` by placing a single `uniformResourceIdentifier` GeneralName ([6] IA5String) in a GeneralNames SEQUENCE under OID 2.5.29.18; generator source available on request.

---

Disclosure: this issue was found and verified with AI assistance (Claude). All lint output above was produced by running `zlint` built from master (7d02b26) against the attached certs before filing.
